backend/src/bundles/users/user.repository.ts

```diff
@@ -13,6 +13,14 @@ class UserRepository implements Repository {
         return Promise.resolve(null);
     }
 
+    public async findByEmail(
+        email: string,
+    ): ReturnType<Repository['findByEmail']> {
+        const user = await this.userModel.query().findOne({ email });
+
+        return user ?? null;
+    }
+
```

Comment:
please double check what does the query return, maybe the ?? null is irrelevent if it already returns null or undefined